### PR TITLE
Ioerror resume

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
Depends on https://github.com/kubevirt/kubevirt/issues/5859

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: tests/io_utils.go
Comment (alicefr) [inline tests/io_utils.go:69]: is the `scsi_debug` module installed in the envrironment?

Comment (xpivarc) [reply tests/io_utils.go:69]: Yes.

**Comment by xpivarc** [comment]
Ready for review

File: tests/storage/storage.go
Comment (alicefr) [inline tests/storage/storage.go:177]: Just a minor remark. You might want to fetch the disk device from the vmi status instead of hardcoding the device

**Comment by alicefr** [comment]
Looks good to me :)

**Comment by aglitke** [comment]
Awesome test!

**Comment by xpivarc** [comment]
@alicefr can you have another look? I had to rebase.
/cc @stu-gott  maybe you can have a look as approver?

**Comment by alicefr** [comment]
@xpivarc for me looks good. The only point I have if you want to hard code the faulty device in the guest (comment: https://github.com/kubevirt/kubevirt/pull/5864/commits/8b02875d82d3167a59bae42262ecd722cdfa219d#r655997875)

File: tests/io_utils.go
Comment (dhiller) [inline tests/io_utils.go:49]: Maybe we might name the first return argument, which would make clear what it returns.

File: tests/io_utils.go
Comment (dhiller) [inline tests/io_utils.go:67]: Same here.

File: tests/io_utils.go
Comment (dhiller) [inline tests/io_utils.go:82]: Maybe a comment with an example on how the output looks like fort this line would help the reader understand what's going on here.

**Review by dhiller — APPROVED** [review]
I only found minor readability nits.

/approve
/hold

Holding for you to consider whether you want to address the comments that are remaining.
Otherwise feel free to unhold.

File: tests/storage/storage.go
Comment (xpivarc) [reply tests/storage/storage.go:179]: I think it should be ok for now since we don't change the os that much. I will have a look into it for future

**Comment by xpivarc** [comment]
@dhiller Added named return variables and example output.

**Review by dhiller — APPROVED** [review]
/unhold

**Comment by dhiller** [comment]
/lgtm cancel

The failures on the two failing lanes have been exactly the same for the last two runs. Could you take a look again please?

**Comment by xpivarc** [comment]
> /lgtm cancel
> 
> The failures on the two failing lanes have been exactly the same for the last two runs. Could you take a look again please?

I thought they are flaky since this is test-only PR. I am going to have a look anyway.

**Comment by xpivarc** [comment]
/retest
@dhiller all of the tests got into [Quarantine](https://github.com/kubevirt/kubevirt/pull/5998). I am not seeing correlation with this PR.

**Review by dhiller — APPROVED** [review]

**Pull request merged**
